src/Hamiltonians/angular_momentum.jl

```diff
@@ -0,0 +1,105 @@
+"""
+    AxialAngularMomentumHO(S; z_dim = 3) <: AbstractHamiltonian
+
+Angular momentum operator for application to Cartesian harmonic oscillator basis,
+see [`HOCartesianEnergyConserved`](@ref) or [`HOCartesianEnergyConservedPerDim`](@ref).
+Represents the projection of angular momentum onto `z`-axis:
+```math
+\\hat{L}_z = i \\hbar \\sum_{j=1}^N \\left( b_{j_x} b_{j_y}^\\dag - b_{j_y} b_{j_x}^\\dag \\right),
+```
+where ``b_{j_x}^\\dag`` and ``b_{j_x}`` are raising and lowering (ladder) operators 
+for the ``j^\\text{th}`` particle in a harmonic oscillator in the ``x`` dimension, 
+and simlarly for ``y``. Their action on a ``N`` particle Fock state that represents 
+a many-body harmonic oscillator is e.g.
+```math
+\\sum_{j=1}^N b_{j_x} b_{j_y}^\\dag = \\sum_{n_x=1}^{M_x} \\sum_{n_y=1}^{M_y} a_{n_x-1,n_y+1}^\\dag a_{n_x, n_y}
+```
+where ``a_n^\\dag`` and ``a_n`` are creation and annihilation operators for the 
+``n^\\text{th}`` single-particle mode, indexed by a pair ``n = (n_x, n_y)``.
+
+Argument `S` is a tuple defining the range of Cartesian modes in each dimension. 
+If `S` indicates a 3D system the `z` dimension can be changed by setting `z_dim`; 
+`S` should be be isotropic in the remaining `x`-`y` plane, i.e. must have 
+`S[x_dim] == S[y_dim]`.
+
+Note: This operator does not have a meaningful [`starting_address`](@ref), but will work with
+[`BasisSetRep`](@ref)`(L, addr)` if an appropriate `addr` is supplied, that is, if
+`prod(S) == num_modes(addr)`.
+"""
+struct AxialAngularMomentumHO{D} <: AbstractHamiltonian{ComplexF64}
+    S::NTuple{D,Int64}
+    xyz::NTuple{3,Int64}
+end
```

Comment:
```suggestion
struct AxialAngularMomentumHO{D, A<:AbstractFockAddress} <: AbstractHamiltonian{ComplexF64}
    S::NTuple{D,Int64}
    xyz::NTuple{3,Int64}
    addr::A
end
```

I would prefer properly supporting a `starting_address` rather than providing a fake one. It is about supporting the `AbstractHamiltonians` interface properly. It is easy to do and won't const any CPU time on code execution (and I doubt that it will cost a measurable amount of compilation time).